Packs/ThreatGrid/Integrations/ThreatGrid/ThreatGrid.py

```diff
@@ -1,1233 +1,1520 @@
+    elif api_score < 50:
+        score = Common.DBotScore.GOOD
+
+    return Common.DBotScore(
+        indicator=indicator_value,
+        indicator_type=generic_command_name,
+        integration_name="ThreatGrid",
+        reliability=reliability,
+        score=score,
+    )
+
+
+def reputation_command(
+    client: Client,
+    args: Dict[str, Any],
+) -> Union[List[CommandResults], CommandResults]:
+    """
+    Generic reputation command that returns information about Files/IPs/URLs/Domains.
+    Args:
+        client (Client): Cisco ThreatGrid API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+    Returns:
+        List[CommandResults]: Indicator for every file_hash
+    """
+    generic_command_name = args["command_name"]
+    command_args = argToList(args[generic_command_name])
+    reliability = args["reliability"]
+    sample_id = ""
+    score = 0
+    command_results = []
+
+    for command_arg in command_args:
+        response = client.search_submission_request(
+            query=command_arg,
+            state="succ",
+            sort_by="analyzed_at",
+        )
+        if response["data"]["current_item_count"] == 0:
+            return CommandResults(readable_output="Unknown")
+
+        sample_details = response["data"]["items"][0]["item"]
+        sample_analysis_date = dict_safe_get(
+            sample_details, ["analysis", "metadata", "sandcastle_env", "analysis_end"])
+
+        if not validate_days_diff(sample_analysis_date):
+            return CommandResults(readable_output="Unknown")
+
+        sample_id = sample_details["sample"]
+        score = sample_details["analysis"]["threat_score"]
+
+        dbot_score = get_dbotscore(score, generic_command_name, command_arg, reliability)
+
+        reputation_helper_command: Callable = REPUTATION_TYPE_TO_FUNCTION[generic_command_name]
+        kwargs = {
+            'client': client,
+            'command_arg': command_arg,
+            'sample_id': sample_id,
+            'dbot_score': dbot_score,
+            'sample_details': sample_details,
+        }
+        command_results.append(reputation_helper_command(**kwargs))
+
+    return command_results
+
+
+""" HELPER FUNCTIONS """
+
+
+def validate_days_diff(sample_analysis_date: str) -> bool:
+    """Validate days diff between today and the specified
+        date is no more than 14 days.
+
+    Args:
+        sample_analysis_date (str): The specified date.
+
+    Returns:
+        bool: Return True is diff smaller than 14.
     """
-    if artifacts:
-        for artifact in artifacts.values():
-            if demisto.get(artifact, 'antivirus.virustotal'):
-                vt = demisto.get(artifact, 'antivirus.virustotal')
-                return {
-                    'Hits': demisto.get(vt, 'hits'),
-                    'Engines': demisto.get(vt, 'engines')
-                }
+    analysis_date = str(sample_analysis_date).split("T")[0]
+    today_date = str(datetime.now()).split(" ")[0]
```

Comment:
please wrap with a try-except block, catching IndexError and raising DemistoException that the time doesn't match the expected format.